app/controllers/idv/welcome_controller.rb

```diff
@@ -12,7 +12,7 @@ def show
       Funnel::DocAuth::RegisterStep.new(current_user.id, sp_session[:issuer]).
         call('welcome', :view, true)
 
-      @presenter = Idv::WelcomePresenter.new(decorated_sp_session)
+      @presenter = Idv::WelcomePresenter.new(decorated_sp_session, current_user)
```

Comment:
If we expected `view_context` on `ServiceProviderSession` and `NullServiceProviderSession` (the two classes that `decorated_sp_session` could be), we could get access to `current_user` that way?

(alternatively, we could define `#current_user` on those two classes and expose that instead of having to pass around another param here?)

Reply:
Made `#view_context` public.

Reply:
Sorry to chime in late here, but I don't think I agree with making `view_context` public. I prefer having `ServiceProviderSession` and `NullServiceProviderSession` define a `current_user` method so we can reduce our reliance on long chains of method calls like `decorated_sp_session.view_context.current_user`

Reply:
Moved everything around.